frontend/app/components/menu.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { Link } from '@remix-run/react';
+
+import { useTranslation } from 'react-i18next';
+
+interface MenuItem {
+  id: string;
+  key: string;
+  value: string;
+  path: string;
+  showIcon: boolean;
+}
+interface MenuProps {
+  menuList: MenuItem[];
+}
+export function Menu({ menuList }: MenuProps) {
+  const { t } = useTranslation('gcweb');
+  const [showDropdown, setShowDropdown] = useState<boolean>(false);
+  const dropdown = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    // Hide dropdown when click outside
+    if (!showDropdown) return;
+    function handleClick(event: MouseEvent) {
+      if (dropdown.current && !dropdown.current.contains(event.target as Node)) {
+        setShowDropdown(false);
+      }
+    }
+    window.addEventListener('click', handleClick);
+    return () => window.removeEventListener('click', handleClick);
+  }, [showDropdown]);
+  useEffect(() => {
+    if (!showDropdown) return;
+    function handleEsc(event: KeyboardEvent) {
+      if (event.key === 'Escape') {
+        setShowDropdown(false);
+      }
+    }
+    window.addEventListener('keydown', handleEsc);
+    return () => window.removeEventListener('keydown', handleEsc);
+  }, [showDropdown]);
+  return (
+    <nav ref={dropdown} data-testid="menu-id">
+      <button
+        id="dropdownNavbarLink"
+        onClick={() => setShowDropdown((e) => !e)}
```

Comment:
**Nitpick:** using `e` here confused me because it's usually reserved for naming the *event*. My suggestion would be to name it something like `currentState` or something like that.